standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -1624,6 +1632,49 @@ public void alter_database(final String dbName, final Database newDB) throws TEx
     }
   }
 
+  private boolean isAlterReplSpecific(Database oldDb, Database newDb) {
+    Map<String, String> oldDbProp = oldDb.getParameters();
+    Map<String, String> newDbProp = newDb.getParameters();
+    List<String> replDbProps = Arrays.stream(ReplConst.class.getDeclaredFields())
+            .filter(field -> Modifier.isStatic(field.getModifiers()))
+            .map(field -> {
+              try {
+                String prop = (String) field.get(String.class);
+                return prop.replace("\"", "");
+              } catch (IllegalAccessException e) {
+                throw new RuntimeException(e);
+              }
+            }).collect(Collectors.toList());
+    boolean replSpecific = false;
+    if (newDbProp != null) {
+      for (Map.Entry<String, String> prop : newDbProp.entrySet()) {
+        String propName = prop.getKey().replace("\"", "");
+        String oldValue = (oldDbProp == null) ? null : oldDbProp.get(prop.getKey());
+        if (!prop.getValue().equals(oldValue)) {
+          if (propName.startsWith(ReplConst.BOOTSTRAP_DUMP_STATE_KEY_PREFIX) || replDbProps.contains(propName)) {
+            replSpecific = true;
+          } else {
+            return false;
```

Comment:
Is the intent to replace/skip all REPL related properties? If so, should the logic be different? As in, why bailing out on first non-repl case?